File: Algorithms/C++/9-Palindrome_Number.cpp
// Runtime: 12 ms, faster than 72.67% of C++ online submissions for Palindrome Number.
// Memory Usage: 8.1 MB, less than 94.55% of C++ online submissions for Palindrome Number.
class Solution {
public:
    bool isPalindrome(int x) {
        long int reversed_num = 0, temp = x;
        while (temp > 0) {
            reversed_num = reversed_num * 10 + temp % 10;
            temp /= 10;
        }
        return reversed_num == x;
    }
};

File: Algorithms/Python3.x/9-Palindrome_Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        return str(x) == str(x)[::-1]


class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        p, res = x, 0
        while p:
            res = res * 10 + p % 10
            p = int(p / 10)
        return res == x
